src/utils/guards/valid-manager.guard.ts

```diff
@@ -0,0 +1,33 @@
+import { User } from '../../modules/users/entities/user.entity';
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { Role } from '../enums/roles.enum';
+
+@Injectable()
+export class ValidManager implements CanActivate {
+  constructor(private reflector: Reflector) {}
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const rolesKey = 'roles';
```

Comment:
I can improve this, It could be better to create another file with these kind of constants.